src/iris/app_stats.py

```diff
@@ -84,116 +68,226 @@ def calculate_app_stats(app, connection, cursor, fields_filter=None):
+                            AND `message`.`application_id` = %(application_id)s
+                            AND `message`.`created` > %(date_variable)s - INTERVAL 1 WEEK
+                            AND `message`.`created` < %(date_variable)s
+                        GROUP BY thisStatus, `mode`.`name`''', query_data)
+        logger.info('App Stats (%s) mode status query took %s seconds', app['name'], round(time.time() - start, 2))
+
+        for mode, status, count in cursor:
+            if isinstance(status, str) and status.isdigit():
+                status = int(status)
+            mode_status[mode][status] = count
+
+        for mode in mode_stats_types:
+            status_counts = mode_status[mode]
+            overall = float(sum(status_counts.values()))
+            if overall > 0:
+                fail_pct = round(
+                    (sum(status_counts.pop(key, 0) for key in mode_stats_types[mode]['fail']) / overall) * 100,
+                    2)
+                success_pct = round(
+                    (sum(status_counts.pop(key, 0) for key in mode_stats_types[mode]['success']) / overall) * 100, 2)
+                other_pct = round(sum(status_counts.values()) / overall * 100, 2)
+            else:
+                fail_pct = success_pct = other_pct = None
+
+            key = 'pct_%s_success_last_week' % mode
+            if stats.get(key):
+                stats['pct_%s_success_last_week' % mode].append({unix_date: success_pct})
+                stats['pct_%s_fail_last_week' % mode].append({unix_date: fail_pct})
+                stats['pct_%s_other_last_week' % mode].append({unix_date: other_pct})
+            else:
+                stats['pct_%s_success_last_week' % mode] = []
+                stats['pct_%s_fail_last_week' % mode] = []
+                stats['pct_%s_other_last_week' % mode] = []
+
+                stats['pct_%s_success_last_week' % mode].append({unix_date: success_pct})
+                stats['pct_%s_fail_last_week' % mode].append({unix_date: fail_pct})
+                stats['pct_%s_other_last_week' % mode].append({unix_date: other_pct})
+
+        # Get counts of messages sent per mode
+        cursor.execute('''SELECT `mode`.`name`, `msg_count` FROM
+                            (SELECT `mode_id`, COUNT(*) AS `msg_count` FROM `message`
+                            USE INDEX (ix_message_sent)
+                            WHERE `sent` > %(date_variable)s - INTERVAL 1 WEEK
+                            AND `sent` < %(date_variable)s
+                            AND `application_id` = %(application_id)s
+                            GROUP BY `mode_id`) `counts`
+                        JOIN `mode` ON `mode`.`id` = `counts`.`mode_id`''',
+                       query_data)
+
+        for row in cursor:
+            key = 'total_%s_sent_last_week' % row[0]
+            result = row[1]
+
+            if stats.get(key):
+                stats[key].appennd({unix_date: result})
```

Comment:
`appennd` is a typo here and on line 169. We should fix this typo, make sure the rest of the script has been tested thoroughly, and try to patch the gaps in our e2e testing to make sure we don't miss things like this.